pkg/client/client.go

```diff
@@ -63,15 +66,18 @@ func (n *NutanixClientHelper) GetClientFromEnvironment(ctx context.Context, nuta
 	// If PrismCentral is set, add the required env provider
 	prismCentralInfo := nutanixCluster.Spec.PrismCentral
 	if prismCentralInfo != nil {
-		if prismCentralInfo.Address == "" {
-			return nil, fmt.Errorf("cannot get credentials if Prism Address is not set")
+		address, err := validateAndSanitizePrismCentralInfoAddress(prismCentralInfo.Address)
+		if err != nil {
+			return nil, err
 		}
+		prismCentralInfo.Address = address
 		if prismCentralInfo.Port == 0 {
 			return nil, fmt.Errorf("cannot get credentials if Prism Port is not set")
 		}
-		credentialRef := prismCentralInfo.CredentialRef
-		if credentialRef == nil {
-			return nil, fmt.Errorf("credentialRef must be set on prismCentral attribute for cluster %s in namespace %s", nutanixCluster.Name, nutanixCluster.Namespace)
+		credentialRef, err := GetCredentialRefForCluster(nutanixCluster)
```

Comment:
line no 75: we can use ErrPrismAddressNotSet in return

Reply:
I think the error here is more specific, but i'll move this error out to a var for readability